LaTeX/LaTeX.sublime-syntax

```diff
@@ -1656,3 +1657,137 @@ contexts:
        3: punctuation.definition.group.brace.begin.latex
        4: entity.name.function.frame.latex
        5: punctuation.definition.group.brace.end.latex
+
+  # support for array package
+  pkgarray:
+    - match: |-
+        (?x)
+        (
+          (\\)newcolumntype
+        )
+        (?:
+          (\{)
+          (
+            (?:\\[A-Za-z@]+)
+            | (?:[A-Za-z@])
+          )
+          (\})
+        )
+        (?:(\[)(?:[^\]]*)(\]))*
+        (\{)
+      captures:
+        1: support.function.newcolumntype.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+        4: support.function.latex entity.name.newcolumntype.latex
+        5: punctuation.definition.group.brace.end.latex
+        6: punctuation.definition.group.bracket.begin.latex
+        7: punctuation.definition.group.bracket.end.latex
+        8: punctuation.definition.group.brace.begin.latex
+      push:
+        - meta_include_prototype: false
+        - meta_scope: meta.function.newcolumntype.latex
+        - match: '\}'
+          scope: punctuation.definition.group.brace.end.latex
+          pop: true
+        - include: array-preamble
+
+
+    - match: |-
+        (?x)
+        ((\\)begin)(\{)(tabular)(\})
+        (?:(\[)(?:b|c|t)(\]))?
+      captures:
+        1: support.function.begin.latex keyword.control.flow.begin.latex
+        2: punctuation.definition.backslash.latex
+        3: punctuation.definition.group.brace.begin.latex
+        4: variable.parameter.function.latex
+        5: punctuation.definition.group.brace.end.latex
+        6: punctuation.definition.group.bracket.begin.latex
+        7: punctuation.definition.group.bracket.end.latex
+      push:
+        - meta_scope: meta.environment.tabular.latex
```

Comment:
@wbond From my point of view the only remaining problem is, that this scope does not span correctly over the following set-chain. E.g. you have

text.tex.latex **meta.environment.tabular.latex meta.environment.tabular.latex** meta.function.column-spec.latex punctuation.definition.group.brace.begin.latex

instead of

text.tex.latex **meta.environment.tabular.latex** meta.function.column-spec.latex punctuation.definition.group.brace.begin.latex

at the braces. Hence it is not ready to pull, but only a little work left.